global.php

```diff
@@ -1018,19 +1019,46 @@
 			$allowable_actions = explode(',', ALLOWABLE_PAGE);
 			if(!in_array($mybb->get_input('action'), $allowable_actions))
 			{
-				error_no_permission();
+				if(!$mybb->get_input('modal'))
```

Comment:
This could be shortened to a temporary true/false variable and a single call instead of three. Not important, but IMO slightly cleaner. Except these 2 small changes it can be merged.